### PR TITLE
Rewrite logging docs.

### PR DESCRIPTION
**Pull request by danielballan**
This borrows substantially from caproto, but it's different enough
that it cannot practically share source with it.

### CODE REVIEW

**Comment by danielballan** [comment]
Closes #1316

**Comment by danielballan** [comment]
Closes #1316

File: doc/source/debugging.rst [hunk 1]
```diff
@@ -1,54 +1,183 @@
+*********************
 Debugging and Logging
-=====================
+*********************
 
-Bluesky uses Python logging framework. See below for a list of the names of the
-loggers it publishes to.
+.. versionchanged:: 1.6.0
+
+   Bluesky's use of Python logging framework has been completely reworked to
```
Comment (jklynch) [inline doc/source/debugging.rst]: "the" Python logging framework?

File: doc/source/debugging.rst [hunk 2]
```diff
@@ -1,54 +1,183 @@
...
+
+.. code-block:: python
+
+   logger.setLevel('DEBUG')
+
+This makes DEBUG-level records *available* to all handlers. Our logstash
+handler, set to ``'INFO'``, will filter out DEBUG-level records.
+
+To globally disable the generation of any log records at or below a certain
+verbosity, which may be helpful for optimising performance, Python provides
```
Comment (jklynch) [inline doc/source/debugging.rst]: Are you practicing your Canadian in case Trump is re-elected? :)

File: doc/source/debugging.rst [hunk 3]
```diff
@@ -101,6 +105,10 @@ additional options---and scoped to the ``'bluesky'`` logger with bluesky's
 :class:`bluesky.log.LogFormatter`. It streamlines common use cases without
 interfering with more sophisticated use cases.
 
+We recommend that facilities using bluesky leave this functions for users and
```
Comment (jklynch) [inline doc/source/debugging.rst]: We recommend that facilities using bluesky leave this "function"? for users and
"configure"? any standardized...

File: doc/source/debugging.rst [hunk 4]
```diff
@@ -27,8 +27,8 @@ This is the recommended standard setup.
    from bluesky import config_bluesky_logging
    config_bluesky_logging()
 
-It will display log records of ``WARNING`` level or higher in the terminal
-(standard out) with a formatting tailored to bluesky.
+It will display ``'bluesky'`` log records of ``WARNING`` level or higher in the
+terminal (standard out) with a formatting tailored to bluesky.
```
Comment (jklynch) [inline doc/source/debugging.rst]: with a "format"?

File: doc/source/debugging.rst (hunk 2 above)
Comment (danielballan) [reply doc/source/debugging.rst]: I honestly think that reading @stuartcampbell's writing has finally got to me. Well done, :gb:.

**Pull request merged**